src/steps/ms-defender/device/__tests__/converters.test.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  createMachineEntity,
+  createMachinesDeviceRelationship,
+  createAccountMachineRelationship,
+} from '../converters';
+import { Entity } from '@jupiterone/integration-sdk-core';
+import { Machine } from '../../../../types';
+const exampleAccountEntity: Entity = {
+  _class: 'Account',
+  _key: 'microsoft_defender_account_id',
+  _type: 'microsoft_defender_account',
+  displayName: 'name',
+  id: '5a721b05-53ed-4ed9-be02-aed28f11edbd',
+};
+
+describe('createMachineEntity', () => {
+  test('properties transferred', () => {
+    const VmMetadata = {
+      vmId: 'string',
+      cloudProvider: 'string',
+      resourceId: 'string',
+      subscriptionId: 'string',
+    };
+    const rawData: Machine = {
+      id: '5a721b05-53ed-4ed9-be02-aed28f11edbd',
+      computerDnsName: 'string',
+      firstSeen: 'string',
+      lastSeen: 'string',
+      classification: 'string',
+      onboardingStatus: 'string',
+      healthStatus: 'string',
+      deviceValue: 'string',
+      agentVersion: 'string',
+      defenderAvStatus: 'string',
+      riskScore: 'string',
+      osPlatform: 'string',
+      osVersion: 'any',
+      osProcessor: 'string',
+      version: 'string',
+      lastIpAddress: 'string',
+      lastExternalIpAddress: 'string',
+      osBuild: 1234,
+      rbacGroupId: 456,
+      rbacGroupName: 'string',
+      exposureLevel: 'string',
+      isAadJoined: true,
+      aadDeviceId: 'any',
+      machineTags: [],
+      osArchitecture: 'string',
+      managedBy: 'string',
+      managedByStatus: 'string',
+      ipAddresses: [],
+      vmMetadata: VmMetadata,
+    };
+    expect(createMachineEntity(rawData)).toEqual({
+      _class: ['HostAgent'],
+      _key: '5a721b05-53ed-4ed9-be02-aed28f11edbd',
+      _rawData: [],
+      _type: 'microsoft_defender_machine',
+      agentVersion: 'string',
+      computerDnsName: 'string',
+      createdOn: undefined,
+      defenderAvStatus: 'string',
+      displayName: 'string',
+      function: ['endpoint-protection', 'vulnerability-detection'],
+      id: '5a721b05-53ed-4ed9-be02-aed28f11edbd',
+      name: 'string',
+      riskScore: 'string',
+    });
+  });
+});
+
+describe('createMachinesDeviceRelationship', () => {
+  test('properties transferred', () => {
+    expect(createMachinesDeviceRelationship(exampleAccountEntity)).toEqual({
+      _class: 'MANAGES',
+      _key: 'microsoft_defender_account_id|manages|FORWARD:_type=user_endPoint:_class=Device,Host:hostname=undefined',
```

Comment:
The device `_key` value seems unreliable, so it may not be unique. Is there no UUID value for the machine data? Isn't the machine `id` used for the `_key`?